### DISCUSSION TITLE
Jacoco ratio de cobertura de test en 0.00

### DISCUSSION THREAD

**Discussion by CarolinaGuercio** [dds-jv/foro#224] [Consultas de herramientas] [answered]
Buenas, estoy teniendo problemas con el **mvn verify** por el ratio de cobertura de test. Se mantiene en 0.00 sin importar cuantos agregue... Alguna configuracion que pueda tener mal? 

![Screenshot from 2021-05-07 20-02-56](https://user-images.githubusercontent.com/29525070/117516370-50659f80-af6f-11eb-8f86-98ff0c1f6484.png)

**Comment by nanderson87** [comment, marked as answer]
Hola @CarolinaGuercio:

1. Si ejecutas el comando `mvn test` cual es el resultado? Te dice que corrieron test  o te dice que se corrieron cero test? Nos podrías adjuntar el resultado de ese comando completo.

**Reply by CarolinaGuercio** [reply]
Dice que corrieron 8 test, Te paso:

![Screenshot from 2021-05-07 21-14-41](https://user-images.githubusercontent.com/29525070/117519288-58c2d800-af79-11eb-934d-e1c2359c584b.png)

**Reply by nanderson87** [reply]
Gracias @CarolinaGuercio. Te sumo algunas consultas más:

1. Existe la posibilidad que tengas una (o más de una) clase dentro del package `macowins` que esten fuera de los sub package venta y modelo? 
2. Modificaste algo del archivo pom.xml?

**Reply by CarolinaGuercio** [reply]
1) Si, hay una clase mas, a la altura de esos dos package que mencionaste. No tiene test esa
2) No

**Reply by nanderson87** [reply]
Genial. Entonces intuyo que el problema viene por ese lado.
Jacoco va a estar verificando que tengas un 60% de cobertura en cada package. SI agregas test en ese clase el verify debería pasar.

**Reply by CarolinaGuercio** [reply]
Si, se ve que era eso! Funciono. Pero por que lo marcaba como 0.00 cuando no pasaba? habia al menos 50% de test

**Reply by nanderson87** [reply]
Porque Jacoco ve la cobertura a nivel de package de forma independiente sin importar si un package esta dentro de otro. Y en cada package ve las clases que tiene adentro.

macowins (con una clase) 
macowins.modelo (con una o más clases)
macowins.venta (con una o más clases)

El package macowins tenia una clase, sin ninguna cobertura, por eso te puso que tenías el 0%. Los otro packages no te dijo nada porque seguramente con la cobertura que tenías cada uno superaba o era igual al 60%

Espero que se haya entendido.

**Reply by CarolinaGuercio** [reply]
Si, perfecto! Gracias por la explicacion. :)
